homeassistant/components/utility_meter/select.py

```diff
@@ -50,8 +51,17 @@ async def async_setup_entry(
     async_add_entities([tariff_select])
 
 
-async def async_setup_platform(hass, conf, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    conf: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the utility meter select."""
+    if discovery_info is None:
+        _LOGGER.error("This platform is only available through discovery")
```

Comment:
```suggestion
        _LOGGER.error(
            "This platform is not available to configure "
            "from 'sensor:' in configuration.yaml"
        )
```

Alternatively, we can remove the log and just return. That's what we normally do.

Reply:
I used similar code to what was in the `sensor` platform:
https://github.com/home-assistant/core/blob/e830032b33a892835482f314e86ceeefd6fa35b1/homeassistant/components/utility_meter/sensor.py#L178-L180

I'm happy to remove the log if you think it's not useful.

Reply:
We can keep it for now since we have it in the oher platforms.